akka-actor-typed-tests/src/test/scala/akka/actor/typed/delivery/ConsumerControllerSpec.scala

```diff
@@ -495,6 +493,11 @@ class ConsumerControllerSpec
       consumerProbe1.expectMessageType[ConsumerController.Delivery[TestConsumer.Job]]
       consumerController ! ConsumerController.Confirmed
 
+      consumerController ! sequencedMessage(producerId, 3, producerControllerProbe.ref)
+      consumerProbe1.expectMessageType[ConsumerController.Delivery[TestConsumer.Job]]
+
+      // now we know that the ConsumerController has received Confirmed for 2,
+      // and 3 is still not confirmed
```

Comment:
The `ConsumerController.Confirmed` was sent but had not arrived before `testKit.stop`. That can be seen in logs, otherwise there would be a `Received Confirmed seqNr [2]`

```
2021-08-18T04:21:17.2613504Z 2021-08-18 04:21:16,975 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging started for test [akka.actor.typed.delivery.ConsumerControllerSpec: ConsumerController must send Ack when stopped] MDC: {}
2021-08-18T04:21:17.2619226Z 2021-08-18 04:21:16,976 DEBUG akka.actor.typed.delivery.ConsumerController  - Cancel timer [Retry] with generation [1] MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2623181Z 2021-08-18 04:21:16,976 DEBUG akka.actor.typed.delivery.ConsumerController  - Received Start, unstash [0] messages. MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2628593Z 2021-08-18 04:21:16,976 TRACE akka.actor.typed.delivery.ConsumerController  - Received first SequencedMessage seqNr [1], delivering to consumer. MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec, producerId=p-15}
2021-08-18T04:21:17.2634431Z 2021-08-18 04:21:16,976 DEBUG akka.actor.typed.delivery.ConsumerController  - Associated with new ProducerController [Actor[akka://ConsumerControllerSpec/system/testProbe-185#866445823]], seqNr [1]. MDC: {akkaAddress=akka://ConsumerControllerSpec, producerId=p-15, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2640970Z 2021-08-18 04:21:16,976 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request with requestUpToSeqNr [20] after first SequencedMessage. MDC: {akkaAddress=akka://ConsumerControllerSpec, producerId=p-15, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2645322Z 2021-08-18 04:21:16,976 TRACE akka.actor.typed.delivery.ConsumerController  - Received Confirmed seqNr [1] from consumer, stashed size [0]. MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2649969Z 2021-08-18 04:21:16,976 DEBUG akka.actor.typed.delivery.ConsumerController  - Sending Request after first with confirmedSeqNr [1], requestUpToSeqNr [20]. MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2654680Z 2021-08-18 04:21:16,976 TRACE akka.actor.typed.delivery.ConsumerController  - Received SequencedMessage seqNr [2], delivering to consumer. MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec, producerId=p-15}
2021-08-18T04:21:17.2659953Z 2021-08-18 04:21:16,976 DEBUG akka.actor.typed.delivery.ConsumerController  - Cancel all timers MDC: {akkaAddress=akka://ConsumerControllerSpec, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec}
2021-08-18T04:21:17.2670613Z 2021-08-18 04:21:16,976 INFO  akka.actor.LocalActorRef akkaDeadLetter - Message [akka.actor.typed.delivery.ConsumerController$Confirmed$] to Actor[akka://ConsumerControllerSpec/user/consumerController-15#-606470781] was not delivered. [2] dead letters encountered. If this is not an expected behavior then Actor[akka://ConsumerControllerSpec/user/consumerController-15#-606470781] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'. MDC: {akkaAddress=akka://ConsumerControllerSpec, sourceThread=ConsumerControllerSpec-akka.actor.default-dispatcher-12, akkaSource=akka://ConsumerControllerSpec/user/consumerController-15, sourceActorSystem=ConsumerControllerSpec, akkaMessageClass=akka.actor.typed.delivery.ConsumerController$Confirmed$, akkaTimestamp=04:21:16.976UTC}
```